lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2557,8 +2557,35 @@ InterfaceTypeRequest::evaluate(Evaluator &eval, ValueDecl *D) const {
       AFD->getParameters()->getParams(argTy);
 
       maybeAddParameterIsolation(infoBuilder, argTy);
-      infoBuilder = infoBuilder.withAsync(AFD->hasAsync());
+
+      auto &ctx = AFD->getASTContext();
+
+      // Check for explicit global actor attribute
+      auto isolationFromAttr = getIsolationFromAttributes(AFD, false);
```

Comment:
Regarding `false` argument here is the idea that `getIsolationFromAttributes` would still be called from `ActorIsolationRequest` and be diagnosed there?  If so, what is the plan to address that in the future because we'd like to remove `getIsolationFromAttributes` from `ActorIsolationRequest` right?

Reply:
> we'd like to remove getIsolationFromAttributes from ActorIsolationRequest right?

I thought the eventual goal was to directly call `ActorIsolationRequest` from `InterfaceTypeRequest` after we move the diagnostics that require the interface type out from `ActorIsolationRequest`.

Reply:
I think the question is still valid though - if we want to call ActorIsolationRequest from InterfaceTypeRequest without diagnostics diagnostics would have to go somewhere.

Reply:
>  but there is something related to getOverridenDeclarations that makes it recursive when called from interface type?
Yes this is the second reason we can't call `ActorIsolationRequest` directly from `InterfaceTypeRequest`

Reply:
Just to be clear I don't think this is a blocker, just trying to understand what the plan is for that because if we don't have test-cases for some situations we'd just loose diagnostics without noticing...